vsphere/datadog_checks/vsphere/api.py

```diff
@@ -72,10 +72,10 @@ def smart_connect(self):
         """
         context = None
         if not self.config.ssl_verify:
-            context = ssl.SSLContext(ssl.PROTOCOL_TLS)
+            context = ssl.SSLContext(PROTOCOL_TLS_CLIENT)
```

Comment:
What motivates this change? `ssl.PROTOCOL_TLS` is OK on both Python 2 and 3. 🤔 Using `PROTOCOL_TLS_CLIENT` has extra behavior we don't want in the case of `ssl_verify: false`:

> The protocol enables `CERT_REQUIRED` and `check_hostname` by default.

```suggestion
            context = ssl.SSLContext(ssl.PROTOCOL_TLS)
```

Reply:
Updated in favour of using `ssl.PROTOCOL_TLS`.